redwood-treehouse-host/src/androidMain/kotlin/app/cash/redwood/treehouse/AndroidTreehouseDispatchers.kt

```diff
@@ -49,3 +51,11 @@ internal class AndroidTreehouseDispatchers : TreehouseDispatchers {
     executorService.shutdown()
   }
 }
+
+@OptIn(ExperimentalCoroutinesApi::class) // CloseableCoroutineDispatcher is experimental.
+internal fun ziplineLoaderDispatcher(): CloseableCoroutineDispatcher {
+  val executorService = Executors.newSingleThreadExecutor { runnable ->
```

Comment:
Does this need a dedicated thread? Shouldn't `Disaptchers.IO` be sufficient? Same for iOS.

Reply:
Unfortunately Dispatchers.IO isn’t thread-confined, and I need that for SQLite.

Reply:
I might be able to borrow a thread from that pool though. Somehow.

Reply:
You can limit the parallelism to 1 if concurrency/parallelism is the problem, but that doesn't guarantee the same thread is used.

Reply:
(In follow-up, as I do.)

Reply:
https://github.com/cashapp/redwood/pull/2188/files